src/d/actor/d_a_player_dproc.inc

```diff
@@ -205,483 +513,1830 @@ BOOL daPy_lk_c::setGetDemo() {
-    /* Nonmatching */
     static const Vec ripple_scale = {0.75f, 0.75f, 0.75f};
+
+    mFrameCtrlUnder[UNDER_MOVE0_e].setRate(daPy_HIO_restart_c0::m.field_0xC);
+    voiceStart(22);
+    mDoAud_bgmStop(0);
+    mDoAud_bgmStart(JA_BGM_DIE_LINK);
+    dCam_getBody()->StartEventCamera(9, fopAcM_GetID(this), 0);
+    if (dComIfGp_checkPlayerStatus0(0, daPyStts0_SWIM_e)) {
+        dComIfGp_particle_setShipTail(
+            dPa_name::ID_COMMON_003F, &current.pos, NULL, (cXyz *)&ripple_scale,
+            0xff, &dPa_control_c::mSingleRippleEcallBack, -1);
+        m34D6 = 1;
+    }
+    return;
 }
 
 /* 8012F38C-8012F3B8       .text checkGameOverStart__9daPy_lk_cFv */
 BOOL daPy_lk_c::checkGameOverStart() {
-    /* Nonmatching */
+    if ((mCurProc == daPyProc_DEMO_DEAD_e) && (m35A0 < 0.38f)) {
+        return true;
+    }
+    return false;
 }
 
 /* 8012F3B8-8012F5A8       .text dProcDead_init__9daPy_lk_cFv */
 BOOL daPy_lk_c::dProcDead_init() {
-    /* Nonmatching */
+    daPy_ANM dVar6;
+    float dVar8;
+
+    if (mCurProc == daPyProc_DEMO_DEAD_e) {
+        return true;
+    }
+    int iVar4 = checkShipRideUseItem(0);
+    uint uVar2 = checkModeFlg(ModeFlg_SWIM);
+    uint uVar3 = mNoResetFlg0 & 0x100;
+    if (uVar2 != 0) {
+        swimOutAfter(0);
+    }
+    commonProcInit(daPyProc_DEMO_DEAD_e);
+    if (dComIfGp_event_compulsory(this)) {
+        dProcDead_init_sub();
+        dVar8 = daPy_HIO_restart_c0::m.field_0xC;
+    } else {
+        dVar8 = 0.0f;
+        m3570 = 1;
+    }
+    mVelocity = 0.0f;
+    m34D4 = 0;
+    if (iVar4 != 0) {
+        dVar6 = ANM_SHIPDIE;
+    } else {
+        if (uVar2 != 0) {
+            dVar6 = ANM_SWIMDIE;
+            onModeFlg(ModeFlg_SWIM);
+            dComIfGp_setPlayerStatus0(0, daPyStts0_SWIM_e);
+            if (uVar3 == 0) {
+                offNoResetFlg0(daPyFlg0_UNK100);
+                dVar8 = 0.0f;
+                m34D4 = 1;
+            } else {
+                m34C2 = 0;
+            }
+        } else {
+            dVar6 = ANM_DIELONG;
+            onModeFlg(ModeFlg_02000000);
+        }
+    }
+    setSingleMoveAnime(dVar6, dVar8, daPy_HIO_restart_c0::m.field_0x10, -1,
+                       daPy_HIO_restart_c0::m.field_0x14);
+    mDamageWaitTimer = 0;
+    m35E4 = 0.0f;
+    m35A0 = 1.0f;
+    if (m34D4 == 0) {
+        speed.y = 0.0f;
+    }
+    gravity = 0.0f;
+    m3574 = 0;
+    initShipRideUseItem(iVar4, 1);
+    m34D6 = 0;
+    if (dVar8 > 0.0f) {
+        dProcDead_init_sub2();
+    }
+    cancelNoDamageMode();
+    return true;
 }
 
 /* 8012F5A8-8012F928       .text dProcDead__9daPy_lk_cFv */
 BOOL daPy_lk_c::dProcDead() {
-    /* Nonmatching */
+    J3DFrameCtrl &frameCtrl = mFrameCtrlUnder[UNDER_MOVE0_e];
+
+    if (dComIfGp_checkPlayerStatus0(0, daPyStts0_SHIP_RIDE_e)) {
+        setShipRidePos(1);
+    }
+    if (((m3570 != 0) && (dComIfGp_event_compulsory(this))) &&
+        (dProcDead_init_sub(), m34D4 == 0)) {
+        dProcDead_init_sub2();
+    }
```

Comment:
The original devs probably didn't use the comma operator to chain things together like this, Ghidra is the only place I've seen that. I think the original devs probably used two if statements like this:
```suggestion
    if (m3570 != 0 && dComIfGp_event_compulsory(this)) {
        dProcDead_init_sub();
        if (m34D4 == 0) {
            dProcDead_init_sub2();
        }
    }
```